### PR TITLE
[#5456] Edit profile username

### LINKED ISSUE #5456: Edit profile picture
Update: Editing username is now working. Leaving this open until editing a picture works.

**Story**
As a user, I want to be able to edit my picture and username so I can maintain my profile updated both on mobile and desktop.

**Current behavior** 
The user  can't edit username and/or image:
captura de pantalla 2018-08-10 a la s 3 29 10 p m

**Solution**
Allow the user to edit username and profile. This should be synced between mobile and desktop (a change on any device should be reflected on another device). New UI elements and behavior are introduced.
Designs and anotations >> zpl.io/brE1NdX (Zeplin)
[Prototype here](https://invis.io/C7NGZR7MH25).

user profile - edit pic and name

**Platforms**
MacOS & Linux.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #5456

### CODE REVIEW

**Review by vkjr — APPROVED** [review]

**Review by rasom — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs [hunk 1]
```diff
@@ -2,27 +2,31 @@
...
   [react/view styles/profile-badge
    [react/image {:source {:uri photo-path}
                  :style  styles/profile-photo}]
-   [react/text {:style           styles/profile-user-name
-                :font           :medium
-                :number-of-lines 1}
-    name]])
+   (if editing?
+     [react/text-input {:auto-focus    true
+                        :default-value name
```
Comment (vitvly) [inline src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: It's not this line's problem frankly, but neither `defaultValue` nor `value` properties are support in RN-desktop yet.

Edit: `value` is supported, my bad.

Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: Can be temporarily replaced by `:placeholder` I guess

**Comment by chadyj** [comment]
Didn't work for me :(

The "edit" button looked good, and worked. But after tapping it I can't edit the text, and the name isn't shown fully.

Edit button:
screen shot 2018-09-10 at 12 23 30 pm

After clicking edit:
screen shot 2018-09-10 at 12 23 37 pm

I can't actually click in the text and update it.

**Review by vitvly — CHANGES_REQUESTED** [review]
Editing username is not working. Clicking on username in edit mode has no effect. Probably due to a RN-desktop bug

File: src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs (hunk 1 above)
Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: https://github.com/status-im/react-native-desktop/issues/332

Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: Have no idea why TextInput is not clickable though...

Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: @vkjr @MaxRis maybe you might have an idea what's up there

Comment (vkjr) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: When I look at screenshot provided by @chadyj above, first thought is that text-input has no proper size information.

As for missed `TextInput` properties, `value` exists for sure. And I believe it should work fine, I was checking it few weeks ago.

Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: @vkjr I agree and retract my statement about `value` field. Sorry, missed it (

As for missing size information - I was experimenting with adding `flex: 1` to the TextInput component for the profile name, and in both cases (with and without `flex`) user name was visible just fine in my case. Not sure how to reproduce Chad's problem.

Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs:20]: @vkjr I've re-checked things and updated issue title/description for 
https://github.com/status-im/react-native-desktop/issues/332.

Indeed the component becomes clickable if width and height are set (but it should not be necessary). However, what is strange is that for instance in desktop add-new screen input components [(link to code)](https://github.com/status-im/status-react/blob/bd3f9570b30462a60989b9a77c8f4d4514936191/src/status_im/ui/screens/desktop/main/add_new/views.cljs#L34) no width and height need to be set for them to be visible and working properly.

**Comment by chadyj** [comment]
While y'all are working on this can you also take a look at changing the profile pic too?

**Pull request closed without merging**